src/rules/noMisspelledOperatorRule.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as tslint from "tslint";
+import * as ts from "typescript";
+import { SonarRuleMetaData } from "../sonarRule";
+import areEquivalent from "../utils/areEquivalent";
+
+export class Rule extends tslint.Rules.AbstractRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "no-misspelt-operator",
+    description: "Mispelled operators like '=+', '=-' and '=!' should not be used",
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-2757",
+    type: "functionality",
+    typescriptOnly: false,
+  };
+
+  public static formatMessage(identifier: string, suggestedOperator: string) {
+    return `Was "${identifier} ${suggestedOperator}" meant instead?`;
```

Comment:
Lol, @vilchik-elena thanks a lot! The message indeed didn't look very smart to me and for `=!` it made no sense at all, that's why I put a specific one for that case.